openslides_backend/shared/mixins/user_scope_mixin.py

```diff
@@ -179,134 +179,193 @@ def check_permissions_for_scope(
+        """
+        meeting_to_admin_user = defaultdict(set)
+        for meeting_user in (
+            self.datastore.get_many(
                 [
                     GetManyRequest(
                         "meeting_user",
-                        admin_group.get("meeting_user_ids", []),
-                        ["user_id"],
+                        list(meeting_user_ids),
+                        ["user_id", "meeting_id"],
                     )
                 ],
                 lock_result=False,
-            ).get("meeting_user", {})
-            group_ids = meeting_dict.get("group_ids", [])
-            groups = self.datastore.get_many(
-                [
-                    GetManyRequest(
-                        "group", group_ids, ["meeting_user_ids", "permissions"]
-                    )
-                ],
-                lock_result=False,
-            ).get("group", {})
-            for group_id, group in groups.items():
-                meeting_user_ids = group.get("meeting_user_ids", [])
-                group_permissions = group.get("permissions", [])
-                if meeting_user_ids and (
-                    "user.can_manage" in group_permissions
-                    or "user.can_update" in group_permissions
-                ):
-                    admin_meeting_users.update(
-                        self.datastore.get_many(
-                            [
-                                GetManyRequest(
-                                    "meeting_user",
-                                    meeting_user_ids,
-                                    ["user_id"],
-                                )
-                            ],
-                            lock_result=False,
-                        ).get("meeting_user", {})
-                    )
-            if admin_meeting_users:
-                is_admin = False
-                for admin_meeting_user in admin_meeting_users.values():
-                    if admin_meeting_user.get("user_id") == instance_id:
-                        return False
-                    if admin_meeting_user.get("user_id") == self.user_id:
-                        is_admin = True
-                if not is_admin:
-                    return False
-            else:
+            )
+            .get("meeting_user", {})
+            .values()
+        ):
+            meeting_to_admin_user[meeting_user["meeting_id"]].add(
+                meeting_user["user_id"]
+            )
+        return meeting_to_admin_user
+
+    def _analyze_meetings(
+        self, meeting_to_admin_users: dict[int, set[int]], instance_id: int
+    ) -> bool:
+        """
+        Helper function used in method check_for_admin_in_all_meetings.
+        Compares the admin users of all meetings with the ids of requested user and requesting user.
+        Requesting user must be admin in all meetings. Requested user cannot be admin in any.
+        """
```

Comment:
This method contains a very admin-related description, but actually the admin-context only exists because you assume that the data delivered will be admin-related, it would be better to move the calculation of this admin-related data into this function to make it more topical or rename the function to be more generic.